Documentation/add-new-rid.md

```diff
@@ -0,0 +1,146 @@
+  that's already known to work. Or it could be a fork of another Linux
+  distribution which is expected to work
+
+- source-build computes the RID correctly
+
+  Source-build computes the RID, at least on Linux, by looking at
+  `/etc/os-release` and running `${VERSION}.${VERSION_ID}-$(uname
+  -m)`. If the result of this computation itself is wrong (for example
+  a rolling Linux distro doesn't use `VERSION_ID`), the steps in this
+  doc wont be sufficient.
+
+- We just need to add things to the RID graph
+
+  This isn't a brand new platform, just a slight tweak to an existing
+  one.
+
+# Step-by-step fix
+
+This error message is generally from the rolysn build:
+
+```
+error NETSDK1083: The specified RuntimeIdentifier 'foo.bar-x64' is not recognized.
+```
+
+It means that the SDK that's being used to build roslyn doesn't have
+the RID in it's RID graph. We can fix that.
+
+The RID is generally present in 2 places:
+
+1. `RuntimeIdentifierGraph.json` in a built-SDK
+
+2. `Microsoft.NETCore.Platforms/runtime.json` in the source
+   repositories (dotnet/runtime or dotnet/corefx)
+
+Generally fixing the second one is sufficient.
+
+Here's how to fix it.
+
+1. Grab the source code
+
+   For .NET Core (3.1 or earlier):
+
+   ```
+   git clone https://github.com/dotnet/corefx
+   cd corefx
+   git checkout release/$version
+   ```
+
+   For .NET (5 or later):
+
+
+   ```
+   git clone https://github.com/dotnet/runtime
+   cd runtime
+   git checkout release/$version
+   ```
+
+2. Add your new RID to `Microsoft.NETCore.Platforms`'s
+   `runtimeGroups.props` file
+
+   If you are adding a new Linux distro, the chagne might look
+   something like this:
+
+   ```diff
+   --- a/pkg/Microsoft.NETCore.Platforms/runtimeGroups.props
+   +++ b/pkg/Microsoft.NETCore.Platforms/runtimeGroups.props
+   @@ -40,6 +40,11 @@
+          <TreatVersionsAsCompatible>false</TreatVersionsAsCompatible>
+        </RuntimeGroup>
+
+   +    <RuntimeGroup Include="exherbo">
+   +      <Parent>linux</Parent>
+   +      <Architectures>x64</Architectures>
+   +    </RuntimeGroup>
+   +
+        <RuntimeGroup Include="fedora">
+          <Parent>linux</Parent>
+          <Architectures>x64;arm64</Architectures>
+   ```
+
+   If you are just adding a new version, it might look like this:
+
+   ```diff
+   --- a/pkg/Microsoft.NETCore.Platforms/runtimeGroups.props
+   +++ b/pkg/Microsoft.NETCore.Platforms/runtimeGroups.props
+   @@ -43,7 +43,7 @@
+        <RuntimeGroup Include="fedora">
+          <Parent>linux</Parent>
+          <Architectures>x64;arm64</Architectures>
+   -      <Versions>23;24;25;26;27;28;29;30;31;32;33;34</Versions>
+   +      <Versions>23;24;25;26;27;28;29;30;31;32;33;34;35</Versions>
+          <TreatVersionsAsCompatible>false</TreatVersionsAsCompatible>
+        </RuntimeGroup>
+   ```
+
+   Then build with the `/p:UpdateRuntimeFiles=true` flag:
+
+   ```
+   ./build.sh /p:UpdateRuntimeFiles=true
```

Comment:
Maybe this should be 
 ./build.sh -- /p:UpdateRuntimeFiles=true

Reply:
In the dotnet/runtime release/5.0 branch, `./build.sh --help` says:

```
...
Command line arguments starting with '/p:' are passed through to MSBuild.
...
```

Have you run into any cases where `--` was required?

Reply:
Defenitly yes
```
dotnet-v5.0.104-SDK$ ./build.sh --help
usage: ./build.sh [options]
options:
  --with-ref-packages <dir>          use the specified directory of reference packages
  --with-packages <dir>              use the specified directory of previously-built packages
  --with-sdk <dir>                   use the SDK in the specified directory for bootstrapping
use -- to send the remaining arguments to MSBuild
```

```
dotnet-v5.0.104-SDK$ ./build.sh --with-sdk ../bootstrap/.dotnet /p:UpdateRuntimeFiles=true 
Unrecognized argument '/p:UpdateRuntimeFiles=true'
usage: ./build.sh [options]
options:
  --with-ref-packages <dir>          use the specified directory of reference packages
  --with-packages <dir>              use the specified directory of previously-built packages
  --with-sdk <dir>                   use the SDK in the specified directory for bootstrapping
use -- to send the remaining arguments to MSBuild

```

Reply:
Hm. I think we might be talking about different things.

I think you are looking at source-build's `support/tarball/build.sh` (which becomes `./build.sh` when you create the tarball). That needs `--` to separate normal arguments from msbuild arguments.

The step here is for passing the `/p:UpdateRuntimeFiles` flag to dotnet/runtime's build.sh, which doesn't need that: https://github.com/dotnet/runtime/blob/fb2db274ae16120b16c64167cc00538b2acaf155/eng/common/build.sh#L40

Do you know where I could have made that clearer in this doc?

Reply:
Thank You very match

Reply:
May be after "1. Grab the source code" add something like "(Not this repository) because I completly miss it :(

Reply:
Thanks! Does the new commit make it clearer?